src/services/LimitedService.ts

```diff
@@ -0,0 +1,15 @@
+import { ApiClient } from '../api/client.js';
+import { SteamingApiClient } from '../api/streaming-client.js';
+import { ConcurrencyLimiter } from '../utils/concurrency.js';
+
+import { BaseService } from './BaseService.js';
+
+export abstract class LimitedService extends BaseService {
```

Comment:
Remove this class, it's not needed. 
Use `ConcurrencyLimiter` only if needed and rather use composition.